### PR TITLE
Make kubernetes requirement optional for Example DAGs

### PR DESCRIPTION
**Pull request by kaxil**
Currently, because we import `from kubernetes.client import models as k8s` in example_dag, Users need to install `kubernetes` package. This commit makes the requirement optional

Example error with 2.0.0beta3:
```
[2020-11-19 15:11:09,820] {dagbag.py:297} ERROR - Failed to import: /Users/aharmon/airflow2.0/.direnv/python-3.7.6/lib/python3.7/site-packages/airflow/example_dags/example_kubernetes_executor_config.py
Traceback (most recent call last):
  File "/Users/aharmon/airflow2.0/.direnv/python-3.7.6/lib/python3.7/site-packages/airflow/models/dagbag.py", line 294, in _load_modules_from_file
    loader.exec_module(new_module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/aharmon/airflow2.0/.direnv/python-3.7.6/lib/python3.7/site-packages/airflow/example_dags/example_kubernetes_executor_config.py", line 23, in <module>
    from kubernetes.client import models as k8s
ModuleNotFoundError: No module named 'kubernetes'
```

image

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by ashb — APPROVED** [review]

File: airflow/example_dags/example_kubernetes_executor_config.py [hunk 1]
```diff
@@ -31,130 +30,146 @@
...
-    start_task >> other_ns_task
-    start_task >> sidecar_task
-    start_task >> task_with_template
+        start_task >> volume_task >> third_task
+        start_task >> other_ns_task
+        start_task >> sidecar_task
+        start_task >> task_with_template
+except ImportError as e:
+    log.warning("Could not import DAGs in example_kubernetes_executor_config.py: %s", str(e))
+    log.warning("Install kubernetes dependencies with: pip install apache-airflow['kubernetes']")
```
Comment (XD-DENG) [inline airflow/example_dags/example_kubernetes_executor_config.py]: Shall we use `cncf.kubernetes` instead? Given `kubernetes` is planned to be deprecated in 2.1 
https://github.com/apache/airflow/blob/41cf172d6af1a5406367e179df763ef1992e4108/setup.py#L604

**Review by XD-DENG — CHANGES_REQUESTED** [review]

**Review by potiuk — APPROVED** [review]
OK. but as @XD-DENG mentioned - 'cncf.kubernetes' should be use used instead. We are standardizing on 'extra'  == 'provider' as the "default" choice.

File: airflow/example_dags/example_kubernetes_executor_config.py (hunk 1 above)
Comment (potiuk) [reply airflow/example_dags/example_kubernetes_executor_config.py]: Yep. This is the official one. and it is in the docs.

Comment (kaxil) [reply airflow/example_dags/example_kubernetes_executor_config.py]: fixed in a73dcb846

**Review by XD-DENG — APPROVED** [review]

**Pull request merged**